test/integration/handlers/notification/index.test.js

```diff
@@ -534,10 +590,35 @@ function sleep (seconds) {
 
 const getNotifications = async (fsp, operation, id) => {
   try {
-    const response = await Request.sendRequest(`${getNotificationUrl}/${fsp}/${operation}/${id}`, Fixtures.buildHeaders, Enum.Http.Headers.FSPIOP.SWITCH.value, Enum.Http.Headers.FSPIOP.SWITCH.value)
+    const url = `${getNotificationUrl}/${fsp}/${operation}/${id}`
+    Logger.debug(`getNotifications: ${url}`)
+    const response = await Request.sendRequest(
+      url,
+      Fixtures.buildHeaders, 
+      Enum.Http.Headers.FSPIOP.SWITCH.value, 
+      Enum.Http.Headers.FSPIOP.SWITCH.value
+    )
     return response.data
   } catch (error) {
     Logger.error(error)
     throw error
   }
 }
+
+
+const wrapWithRetries = async (func, remainingRetries, timeout) => {
```

Comment:
I think we already have this function defined above (https://github.com/mojaloop/ml-api-adapter/pull/489/files#diff-29685d84407b6f5be72506edf98c74498557b8c6d2c05a654e7acaa94feb5134R84), can we not generalise it into a utility function?

Or alternatively, we could use a library to handle this --> https://www.npmjs.com/package/p-retry?

Reply:
For simplicity's sake, the wait4 functions are standalone and shouldn't depend on anything, nor share any dependencies with the main code base.